00_Base/src/util/AsyncResponder.ts

```diff
@@ -19,39 +19,26 @@ export class AsyncResponder {
     readonly transactionRepo: SequelizeTransactionEventRepository,
   ) {}
 
-  async send(correlationId: string, data: CommandResult) {
-    const responseUrlEntity =
-      await this.responseUrlRepo.getResponseUrl(correlationId);
-    if (responseUrlEntity) {
-      await this.asyncResponseApi.postAsyncResponse(
-        responseUrlEntity.responseUrl,
-        data,
-      );
-    } else {
-      throw new NotFoundError(
-        'No response url found for correlationId: ' + correlationId,
-      );
-    }
-  }
-
-  async sendChargingProfileResult(
+  async send(
     correlationId: string,
     data:
       | ActiveChargingProfileResult
       | ClearChargingProfileResult
-      | ChargingProfileResult,
+      | ChargingProfileResult
+      | CommandResult,
+    sessionId?: string,
```

Comment:
We pass in sessionId to figure out which party id/country code headers to use?

If so, can we instead of update to accept these headers are params instead of sessionId?

We should move the session look up logic out of this method and into whoever is calling this method.

My reasoning is that we are bleeding OCPP logic into this class when we do not have to (e.g. the reference to transaction repo).

Reply:
Not to mention, it looks like the callers of this method already look up transaction before calling this method so we it seems like we have a redundant look up.